osu.Game/Overlays/Profile/Header/PreviousUsernamesContainer.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics;
+using osu.Game.Users;
+using osuTK;
+using System;
+using System.Linq;
+
+namespace osu.Game.Overlays.Profile.Header
+{
+    public class PreviousUsernamesContainer : Container
+    {
+        private const int duration = 200;
+        private const int margin = 10;
+        private const int width = 350;
+        private const int move_offset = 20;
+
+        public readonly Bindable<User> User = new Bindable<User>();
+
+        private readonly ContentContainer contentContainer;
+
+        public PreviousUsernamesContainer()
+        {
+            HoverIconContainer hoverIcon;
+
+            AutoSizeAxes = Axes.Y;
+            Width = width;
+            Children = new Drawable[]
+            {
+                contentContainer = new ContentContainer(),
+                hoverIcon = new HoverIconContainer(),
+            };
+
+            hoverIcon.ActivateHover += () =>
+            {
+                contentContainer.Show();
+                this.MoveToY(-move_offset, duration, Easing.OutQuint);
+            };
+
+            User.BindValueChanged(onUserChanged);
```

Comment:
Generally, we don't want to be binding in the ctor. All bindable bindings should happen in `LoadComplete()` at the earliest